roles/slurm/templates/notification_scripts/nodes.bash

```diff
@@ -1,23 +1,349 @@
+}
+
+#
+# Catch all function for logging using log levels like in Log4j.
+#
+# Requires 5 ARGS:
+#  1. log_level        Defined explicitly by programmer.
+#  2. ${LINENO}        Bash env var indicating the active line number in the executing script.
+#  3. ${FUNCNAME[0]}   Bash env var indicating the active function in the executing script.
+#  4. (Exit) STATUS    Either defined explicitly by programmer or use Bash env var ${?} for the exit status of the last command.
+#  5  log_message      Defined explicitly by programmer.
 #
-# Compile JSON message payload.
-#
-read -r -d '' message << EOM
-{
-	"type": "mrkdwn",
-	"text": "*The _{{ slurm_cluster_name | capitalize }}_ cluster needs help*:  
-Please check and fix my nodes!  
-The \`sinfo\` command reports:
-\`\`\`
-$(sinfo -o "%P|%a|%D|%T|%N|%E" | tr \" \' | column -t -s '|')
-\`\`\`"
+# When log_level == FATAL the script will be terminated.
+#
+# Example of debug log line (should use EXIT_STATUS = 0 = 'OK'):
+#    log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME[0]:-main}" '0' 'We managed to get this far.'
+#
+# Example of FATAL error with explicit exit status 1 defined by the script: 
+#    log4Bash 'FATAL' "${LINENO}" "${FUNCNAME[0]:-main}" '1' 'We cannot continue because of ... .'
+#
+# Example of executing a command and logging failure with the EXIT_STATUS of that command (= ${?}):
+#    someCommand || log4Bash 'FATAL' "${LINENO}" "${FUNCNAME[0]:-main}" "${?}" 'Failed to execute someCommand.'
+#
+function log4Bash() {
+	local _log_level
+	local _log_level_prio
+	local _status
+	local _problematic_line
+	local _problematic_function
+	local _log_message
+	local _log_line_prefix
+	local _log_line
+	#
+	# Validate params.
+	#
+	if [[ ! "${#}" -eq 5 ]]
+	then
+		echo "WARN: should have passed 5 arguments to ${FUNCNAME[0]}: log_level, LINENO, FUNCNAME, (Exit) STATUS and log_message."
+	fi
+	#
+	# Determine prio.
+	#
+	_log_level="${1}"
+	_log_level_prio="${l4b_log_levels["${_log_level}"]}"
+	_status="${4:-$?}"
+	#
+	# Log message if prio exceeds threshold.
+	#
+	if [[ "${_log_level_prio}" -ge "${l4b_log_level_prio}" ]]
+	then
+		_problematic_line="${2:-'?'}"
+		_problematic_function="${3:-'main'}"
+		_log_message="${5:-'No custom message.'}"
+		#
+		# Some signals erroneously report $LINENO = 1,
+		# but that line contains the shebang and cannot be the one causing problems.
+		#
+		if [[ "${_problematic_line}" -eq 1 ]]
+		then
+			_problematic_line='?'
+		fi
+		#
+		# Format message.
+		#
+		_log_line_prefix=$(printf "%-s %-5s @ L%-s(%-s)>" "${SCRIPT_NAME}" "${_log_level}" "${_problematic_line}" "${_problematic_function}")
+		_log_line="${_log_line_prefix} ${_log_message}"
+		if [[ -n "${mixed_stdouterr:-}" ]]
+		then
+			_log_line="${_log_line} STD[OUT+ERR]: ${mixed_stdouterr}"
+		fi
+		if [[ "${_status}" -ne 0 ]]
+		then
+			_log_line="${_log_line} (Exit status = ${_status})"
+		fi
+		#
+		# Log to STDOUT and syslog
+		#
+		logger --id="${$}" -s "${_log_line}"
```

Comment:
I would suggest you also add `--tag` it helps a lot to find logs later when you want to debug a script.

Reply:
Tags are separate from searching. When searching, you can get way more than what you want. For example, not every `slurm` is connected to notification. But `notification` tag could be.
If I see correctly, then currently, the ID is not set to the default tag (which you wrote that is by default set to `username/effective user id`), but is instead changed to process ID (which is most of the time not what you are looking for or, and not what you need).

But since you said that the line is already containing script name, I think that should fine.